doctorapp/src/views/HomeView.vue

```diff
@@ -1,25 +1,96 @@
 <template>
-  <main>
-    <button @click="logout">Logout</button>
-
-    <hr />
-    <Suspense>
-      <!-- <PodList /> -->
-
-      <template #fallback> Loading pods... </template>
-    </Suspense>
-  </main>
+  <div class="dashboard">
+    <HeaderBase :pfp-src="userProfile.img" :name="userProfile.name" title="DOCTORAPP"></HeaderBase>    
+    <main>
+      <div v-if="appointmentsError" class="error-message">
+        <p>Unable to load appointments. Please check your access permissions or try again later.</p>
+      </div>
+      <div v-else class="appointments-container">
+        <Appointments v-if="appointmentsData" :appointments="appointmentsData" class="appointments"/>
+      </div>
+      <div v-if="profileError" class="error-message">
+        <p>Unable to load profile information. Please check your access permissions or try again later.</p>
+      </div>
+      <Sidebar v-else :name="userProfile.name" :description="userProfile.description" :mbox="userProfile.mbox" :img="userProfile.img" :phone="userProfile.phone" class="personal-information"/>
+    </main>
+  </div>
 </template>
 
 <script setup lang="ts">
-// import PodList from "../components/PodList.vue";
-import router from "@/router";
+import { onMounted, ref } from 'vue';
+import HeaderBase from '@/components/home/header/HeaderBase.vue';
+import Appointments from '@/components/home/Appointments.vue';
+import Sidebar from '@/components/home/Sidebar.vue';
+import type { Session } from '@inrupt/solid-client-authn-browser';
+import { getAppointments, getProfileInfo } from 'loama-controller';
 import { store } from "@/store";
 
 const session = store.session;
+const userProfile = ref({
+  name: 'Default Name',
+  mbox: '',
+  img: '',
+  description: '',
+  phone: ''
+});
+const appointmentsData = ref();
+const appointmentsError = ref(false);
+const profileError = ref(false);
+
+const podUrl = store.usedPod.replace(/\/$/, '');
 
-const logout = async () => {
-  session.logout();
-  router.push("/");
-};
+onMounted(async () => {
+    try {
+      const fetchedUserProfile = await getProfileInfo(session as Session, podUrl);
+      userProfile.value = fetchedUserProfile;
+    } catch (error) {
+      profileError.value = true;
+    }
+
+    try {
+      const appointments = await getAppointments(session as Session, podUrl);
+      appointmentsData.value = appointments;
+    } catch (error) {
+      appointmentsError.value = true;
+    }
```

Comment:
It might be interesting to extract this to it's own functions to improve code readability, but this is an nice to have